sway-core/src/type_system/type_info.rs

```diff
@@ -87,6 +87,13 @@ pub enum TypeInfo {
     Storage {
         fields: Vec<TyStructField>,
     },
+    /// Raw untyped pointers.
+    /// These are represented in memory as u64 but are a different type since pointers only make
+    /// sense in the context they were created in. Users can obtain pointers via standard library
+    /// functions such `alloc` or `stack_ptr`. These functions are implemented using asm blocks
```

Comment:
Maybe only mention the `Vec` example here? `alloc` and `stack_ptr` did not really change in this PR. I know Andrew changed them in his previous PR but not sure if that's something we really want to do just yet.